pkl-core/src/main/java/org/pkl/core/ast/member/TypedPropertyNode.java

```diff
@@ -26,6 +26,7 @@
 /** A property definition that has a type annotation. */
 public final class TypedPropertyNode extends RegularMemberNode {
   @Child private DirectCallNode typeCheckCallNode;
+  private final ObjectMember member;
```

Comment:
If booleans can be tightly packed, the cost concern in `ObjectMember` (above) goes away. In that case, I'd also just grab the single boolean here, because a pointer won't be.